test/test_mysql2.rb

```diff
@@ -0,0 +1,148 @@
+require 'test/unit'
+require 'semian/mysql2'
+require 'toxiproxy'
+require 'timecop'
+
+class TestMysql2 < Test::Unit::TestCase
+  def setup
+    @proxy = Toxiproxy[:semian_test_mysql]
+    Semian.destroy(:mysql_testing)
+  end
+
+  def teardown
+    threads.each { |t| t.kill }
+    @threads = []
+  end
+
+  def test_semian_identifier
+    assert_equal :mysql_foo, FakeMysql.new(semian: {name: 'foo'}).semian_identifier
+    assert_equal :'mysql_localhost:3306', FakeMysql.new.semian_identifier
+    assert_equal :'mysql_127.0.0.1:3306', FakeMysql.new(host: '127.0.0.1').semian_identifier
+    assert_equal :'mysql_example.com:42', FakeMysql.new(host: 'example.com', port: 42).semian_identifier
+  end
+
+  def test_resource_acquisition_for_connect
+    client = connect_to_mysql!
+
+    Semian[:mysql_testing].acquire do
+      assert_raises Mysql2::SemianError do
+        Mysql2::Client.new(semian: {name: :testing, tickets: 1, timeout: 0})
+      end
+    end
+  end
+
+  def test_resource_timeout_on_connect
+    @proxy.downstream(:latency, latency: 500).apply do
+      background { connect_to_mysql! }
+
+      assert_raises Mysql2::SemianError do
+        connect_to_mysql!
+      end
+    end
+  end
+
+  def test_circuit_breaker_on_connect
+    @proxy.downstream(:latency, latency: 500).apply do
+      background { connect_to_mysql! }
+
+      3.times do
+        assert_raises Mysql2::SemianError do
+          connect_to_mysql!
+        end
+      end
+    end
+
+    yield_to_background
+
+    assert_raises Mysql2::SemianError do
```

Comment:
Is there a more specific type for circuit breaker errors? We should embed this information into this error.